osu.Game.Rulesets.Osu/Skinning/Argon/ArgonMainCirclePiece.cs

```diff
@@ -186,25 +182,28 @@ private void updateStateTransforms(DrawableHitObject drawableHitObject, ArmedSta
                         {
                             outerGradient.ResizeTo(OUTER_GRADIENT_SIZE * shrink_size, resize_duration, Easing.OutElasticHalf);
 
-                            if (showFlash)
-                            {
-                                outerGradient
-                                    .FadeColour(Color4.White, 80)
-                                    .Then()
-                                    .FadeOut(flash_in_duration);
-                            }
-                            else
-                            {
-                                outerGradient
-                                    .FadeColour(Color4.White, flash_in_duration * 8)
-                                    .FadeOut(flash_in_duration * 2);
-                            }
+                            outerGradient
+                                .FadeColour(Color4.White, 80)
+                                .Then()
+                                .FadeOut(flash_in_duration);
                         }
 
-                        if (showFlash)
+                        if (configHitLighting.Value)
+                        {
+                            flash.HitLighting = true;
                             flash.FadeTo(1, flash_in_duration, Easing.OutQuint);
 
-                        this.FadeOut(showFlash ? fade_out_time : fade_out_time / 2, Easing.OutQuad);
+                            this.FadeOut(fade_out_time, Easing.OutQuad);
+                        }
+                        else
+                        {
+                            flash.HitLighting = false;
+                            flash.FadeTo(1, flash_in_duration, Easing.OutQuint)
+                                 .Then()
+                                 .FadeOut(flash_in_duration, Easing.OutQuint);
+
+                            this.FadeOut(fade_out_time * 0.8f, Easing.OutQuad);
```

Comment:
Of note, this global fade still needs to be adjusted a bit as it was responsible for allowing the lighting to linger in the first place. It can't just be applied to the flash, because the lighting effect is generated by the cross-play of `flash` and `outerGradient`.